src/auth/MockAuthService.js

```diff
@@ -0,0 +1,273 @@
+    logError: PropTypes.func.isRequired,
+    logInfo: PropTypes.func.isRequired,
+  }).isRequired,
+  // The absence of authenticatedUser means the user is anonymous.
+  authenticatedUser: userPropTypes,
+  // Must be at least a valid user, but may have other fields.
+  hydratedAuthenticatedUser: userPropTypes,
+};
+
+/**
+ * The MockAuthService class uses axios-mock-adapter to wrap it's HttpClients so that they can be
+ * mocked for development and testing.  In an application, it could be used to temporarily override
+ * the "real" auth service:
+ *
+ * ```
+ * import {
+ *   initialize,
+ *   APP_AUTH_INITIALIZED,
+ *   subscribe,
+ *   mergeConfig,
+ * } from '@edx/frontend-platform';
+ * import { MockAuthService } from '@edx/frontend-platform/auth';
+ *
+ * initialize({
+ *   handlers: {
+ *     config: () => {
+ *       mergeConfig({
+ *         authenticatedUser: {
+ *           userId: 'abc123',
+ *           username: 'Mock User',
+ *           roles: [],
+ *           administrator: false,
+ *         },
+ *         hydratedAuthenticatedUser: {
+ *           // Additional user props expected to be returned from a user accounts API, providing
+ *           // additional user details beyond those present in the JWT.  This data is added into
+ *           // the user object if and when hydrateAuthenticatedUser is called.  See documentation
+ *           // for `hydrateAuthenticatedUser` for more details.
+ *         }
+ *       });
+ *     },
+ *   },
+ *   messages: [],
+ *   authService: MockAuthService,
+ * });
+ *
+ * subscribe(APP_AUTH_INITIALIZED, () => {
+ *   // This variable is now a MockAdapter from axios-mock-adapter, allowing onGet, onPost, etc.
+ *   // mocking.  This handler will be called prior to any further initialization.  See the
+ *   // "Application Initialization" phases in the README for call order details.
+ *   const mockAuthenticatedHttpClient = getAuthenticatedHttpClient();
+ *   // Mock calls here.
+ * });
+ * ```
+ *
+ * In a test where you would like to mock out API requests - perhaps from a redux-thunk function -
+ * you could do the following to set up a MockAuthService for your test:
+ *
+ * ```
+ * import { getConfig } from '@edx/frontend-platform';
+ * import { configure, MockAuthService } from '@edx/frontend-platform/auth';
+ *
+ * const mockLoggingService = {
+ *   logInfo: jest.fn(),
+ *   logError: jest.fn(),
+ * };
+ * configure(MockAuthService, { config: getConfig(), loggingService: mockLoggingService });
+ * const mockAuthenticatedHttpClient = getAuthenticatedHttpClient();
+ *
+ * // Mock calls for your tests.  This configuration can be done in any sort of test setup.
+ * ```
+ *
+ * NOTE: The login/logout methods related to redirecting currently maintain their real behaviors.  A
+ * subsequent update to this mock service could allow them to be configured/mocked via the
+ * constructor's config options.
+ *
+ * @implements {AuthService}
+ * @memberof module:Auth
+ */
+class MockAuthService {
```

Comment:
SUGGEST/DISCUSS: NONBLOCKING: Just had a thought. Would we be interested in this mock client not being used in production code ever since that would bypass auth? I guess right now one would still add a dependency on auth, as opposed to a devDependency. And can also use this in prod code I think without any restrictions. Outside the scope of this PR but would it be worth having a separate mock adapter dependency that can be pulled in via devDependency instead? Wonder what you think about that. That lib would house this plus any other needed auth utilities (if they exist) that are for use in dev rather than for shipping code

Reply:
I don't think this would nor should ever be used in production - i.e., it has no place where a user may actually see it.  I can see it being used temporarily during early development when backend APIs aren't yet available.  But even there, that's not a very strong use case.  The primary use case is for writing tests of code that uses frontend-platform/auth to make HTTP requests.  

Any consuming app that would use this would presumably already have frontend-platform as a non-dev dependency already.  It's arguable that this class could be split out from frontend-platform itself, but I'd advocate against putting it in its own repository, as we already have too much repo overhead as is.

Also, because of webpack tree shaking, apps won't incur any overhead for this class being in frontend-platform unless they're actually using it.

Reply:
got it thanks!